public/app/plugins/datasource/loki/live_target.ts

```diff
@@ -0,0 +1,25 @@
+import { AppendingVector, DataFrame, Labels } from '@grafana/data';
+import { Subscription } from 'rxjs';
+
+export interface BufferedData {
+  // Direct Access
+  times: AppendingVector<string>;
+  lines: AppendingVector<string>;
+  labels: AppendingVector<Labels>;
+
+  // Structured
+  frame: DataFrame;
```

Comment:
Wondering why the duplicate structure here having both the values in the dataFrame and outside. Does not seem we return it, so is it just so we do not have to search for the correct field in the dataframe when appending? If so seems like we should create some nice accessor for that (I assume right now you have to do manual search in the fields)